convert2rhel/utils.py

```diff
@@ -66,6 +69,141 @@ class Color(object):
+    function as a child process to the parent process with the intention to
+    execute that in its own process, thus, avoiding cases where libraries would
+    install handlers that could be propagated to the main thread if they do not
+    do a clean and proper clean-up.
+
+    .. note::
+        This decorator is mostly intended to be used when dealing with
+        functions that call directly `rpm` or `yum`.
+
+        In `rpm` (version 4.11.3, RHEL 7), it's know that whenever there is a
+        call to that library, it will install a global signal handler catching
+        different types of signals, but most importantly, the SIGINT (Ctrl + C)
+        one, which causes problem during the conversion, as we can't replace or
+        override that signal as it was initiated in a C library rather than a
+        python library.
+
+        By using this decorator, `rpm` will install the signal handler in the
+        child process and leave the parent one with the original signal
+        handling that is initiated by python itself (or, whatever signal is
+        registered when the conversion starts as well).
+
+    .. warning::
+        This decorator will use a instance of `multiprocessing.Queue` to
+        communicate between the processes, so, if this decorator is being used
+        in a function that should return values, it will need to adapt to the
+        queue workflow, and instead of returning it directly, just put the
+        value inside the queue and the decorator will pop it out once the
+        process finishes.
+
+    :param func: Function attached to the decorator
+    :type func: Callable
+    :return: A internal callable wrapper
+    :rtype: Callable
+    """
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        """
+        Wrapper function to execute and control the function attached to the
+        decorator.
+
+        :param args: Arguments tied to the function
+        :type args: tuple
+        :param kwargs: Named arguments tied to the function
+        :type kwargs: dict
+
+        :raises KeyboardInterrupt: Raises a `KeyboardInterrupt` if a SIGINT is
+            catched during the execution of the child process. This exception
+            will be re-raised to the stack once it is catched.
+        :raises Exception: Raise any general exception that can occurs during
+            the execution of the child process.
+
+        :return: If the Queue is not empty, return anything that on it,
+        otherwise, returns `None`.
+        :rtype: Any
+        """
+        queue = multiprocessing.Queue()
+        process_args = (queue,)
+        if args:
+            args += process_args
+            process_args = args
+
+        # TODO(r0x0d): I don't think that creating the process can cause an
+        # exception that we should handle here, only if we mess something with
+        # the process_args tuple, but that should be know during development
+        # time. In any case, check this later.
```

Comment:
@abadger are you aware if this could raise an exception that we should catch? I don't remind of any.

Reply:
I cannot think of any exception that we should catch either.

Reply:
Cool, will remove the comment then.